tests/units/LogScopesTest.cpp

```diff
@@ -0,0 +1,390 @@
+        TRACE_WARN( "This warning message is invisible!!!" );
+        TRACE_ERR( "This error and higher priority messages are visible" );
+    } while ( false );
+
+    do
+    {
+        TRACE_SCOPE( areg_unit_tests_LogScopeTest_ChangeScopePrioAndSaveConfig_fatal );
+        TRACE_DBG( "This debug message is invisible!!!" );
+        TRACE_INFO( "This info message is invisible!!!" );
+        TRACE_WARN( "This warning message is invisible!!!" );
+        TRACE_ERR( "This error message is invisible!!!" );
+        TRACE_FATAL( "This fatal error priority messages is visible" );
+
+    } while ( false );
+
+    do
+    {
+        TRACE_SCOPE( areg_unit_tests_LogScopeTest_ChangeScopePrioAndSaveConfig_noscope );
+        TRACE_DBG( "This log has no scope output" );
+        TRACE_INFO( "Check the log output by scope ID" );
+        TRACE_WARN( "Enabling scope output and you'll see only exit" );
+        SCOPE_PRIORITY_CHANGE( areg_unit_tests_LogScopeTest_ChangeScopePrioAndSaveConfig_noscope, PRIO_LOG_ALL );
+    } while ( false );
+
+    // bring back the prio settings:
+    SCOPE_PRIORITY_CHANGE( areg_unit_tests_LogScopeTest_ChangeScopePrioAndSaveConfig_noscope, PRIO_NOSCOPES( PRIO_WARNING ) );
+
+    do
+    {
+        TRACE_SCOPE( areg_unit_tests_LogScopeTest_ChangeScopePrioAndSaveConfig_nolog );
+        TRACE_DBG( "This debug message is invisible!!!" );
+        TRACE_INFO( "This info message is invisible!!!" );
+        TRACE_WARN( "This warning message is invisible!!!" );
+        TRACE_ERR( "This error message is invisible!!!" );
+        TRACE_FATAL( "This fatal message is invisible!!!" );
+    } while ( false );
+
+    do
+    {
+        TRACE_SCOPE( areg_unit_tests_LogScopeTest_ChangeScopePrioAndSaveConfig );
+        TRACE_DBG( "Saving current state in the [ %s ] file", testConfig.getString( ) );
+        ASSERT_TRUE( NETrace::saveLogging( testConfig ) );
+        TRACE_DBG( "Successfully saved [ %s ] file", testConfig.getString( ) );
+    } while ( false );
+
+    TRACER_STOP_LOGGING( );
+    ASSERT_FALSE( IS_TRACE_STARTED( ) );
+}
+
+/**
+ * \brief   This test is checking the grouping by priority.
+ **/
+DEF_TRACE_SCOPE( areg_unit_tests_LogScopeTest_ScopePriorityGroupping_errLeaf1 );
+DEF_TRACE_SCOPE( areg_unit_tests_LogScopeTest_ScopePriorityGroupping_errLeaf2 );
+DEF_TRACE_SCOPE( areg_unit_tests_LogScopeTest_ScopePriorityGroupping_fatalNode1_leaf );
+DEF_TRACE_SCOPE( areg_unit_tests_LogScopeTest_ScopePriorityGroupping_fatalNode2_leaf );
+DEF_TRACE_SCOPE( areg_unit_tests_LogScopeTest_ScopePriorityGroupping_warnNode1_leaf );
+DEF_TRACE_SCOPE( areg_unit_tests_LogScopeTest_ScopePriorityGroupping_warnNode2_leaf );
+DEF_TRACE_SCOPE( areg_unit_tests_LogScopeTest_ScopePriorityGroupping_infoNode1_noScope );
+DEF_TRACE_SCOPE( areg_unit_tests_LogScopeTest_ScopePriorityGroupping_infoNode2_noScope );
+DEF_TRACE_SCOPE( areg_unit_tests_LogScopeTest_ScopePriorityGroupping_information );
+TEST( LogScopeTest, ScopePriorityGroupping )
+{
```

Comment:
In this test would be good to save the scope priority and later compare the values when load config file.